File: P007.py
import math

def prime(p):
    q = math.sqrt(p)
    if p % 2 == 0:
        return False
    i = 3
    while i <= q:
        if p % i == 0:
            return False
        i += 2
    return True

j = 1
p = 0
while p < 10001:
    j += 1
    if prime(j):
        p += 1


print(j)

File: C/P007.c
#include <stdio.h>
#include <math.h>

int IsPrime(int p){
    float q = sqrt(p);
    if (p % 2 == 0){
        return 0;
    }
    int i = 3;
    while (i <= q){
        if (p  % i == 0){
            return 0;
        }
        i += 2; 
    }
    return 1;
}

int main(int argc, char const *argv[]){
    int j = 1, p = 0;
    while (p < 10001){
        if (IsPrime(j)){
            p +=1;
        }
        j += 1;
    }
    printf("%d\n", j);
    return 0;
}
